### DISCUSSION TITLE
Access raster:bands in ImageData

### DISCUSSION THREAD

**Discussion by fabricebrito** [developmentseed/titiler#649] [General]
This is potentially a long shot but still worth asking :-)

If a custom algorithm could access the mean and stddev of the full image, several algorithms can be implemented in Titiler allowing for very cool use-cases.

Is there a small chance of getting this somewhat?

Example of STAC Asset raster:bands information: 

```json
"raster:bands": [
        {
          "data_type": "uint16",
          "spatial_resolution": 20,
          "nodata": 0,
          "statistics": {
            "minimum": 1,
            "maximum": 16842,
            "mean": 2634.1490243416,
            "stddev": 2634.1490243416,
            "valid_percent": 99.999
          },
          "unit": "W/m²/sr/μm"
        }
      ]
```

**Comment by vincentsarago** [comment]
@fabricebrito interesting, in theory we already forward `min/max` in the ImageData (https://github.com/cogeotiff/rio-tiler/blob/b6b077d2f8e3d30abb95b3f7874ac9253ef5df3b/rio_tiler/reader.py#L235-L243) because we use it to automatically rescale the data when we need.

That's said we could totally extend this to other GDAL usual statistics (feel free to start a PR 🙏 )

**Reply by vincentsarago** [reply]
oh You're showing a STAC example. I've opened https://github.com/cogeotiff/rio-tiler/issues/571 but didn't work on it yet

**Reply by fabricebrito** [reply]
Yes, it's a STAC example and I'm targeting providing a tile-based processing using the full scene stats.
The outcome of https://github.com/cogeotiff/rio-tiler/issues/571 would lead to the tile raster:bands information rather than the full scene right?

**Reply by vincentsarago** [reply]
yeah but maybe we can forward whatever things are in the `asset` dict 😄

**Reply by fabricebrito** [reply]
That would be great!

**Reply by fabricebrito** [reply]
@vincentsarago about extending `"STATISTICS_MINIMUM"` and `"STATISTICS_MAXIMUM"` we can extend this easily but discussing with @emmanuelmathot, he says we don't see these values set often so going for the STAC information looks like the best shot

**Reply by vincentsarago** [reply]
it should be fixed in next rio-tiler major release https://github.com/cogeotiff/rio-tiler/pull/611 🥳
